src/Command/CronCommand.php

```diff
@@ -0,0 +1,116 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\CronJobBundle\Command;
+
+use Becklyn\CronJobBundle\Cron\CronJobRegistry;
+use Becklyn\CronJobBundle\Data\CronStatus;
+use Becklyn\CronJobBundle\Data\WrappedJob;
+use Becklyn\CronJobBundle\Model\CronModel;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CronCommand extends Command
+{
+    public static $defaultName = "cron:run";
+
+
+    /**
+     * @var CronJobRegistry
+     */
+    private $registry;
+
+
+    /**
+     * @var CronModel
+     */
+    private $logModel;
+
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+
+    public function __construct (CronJobRegistry $registry, CronModel $model, LoggerInterface $logger)
+    {
+        parent::__construct();
+        $this->registry = $registry;
+        $this->logModel = $model;
+        $this->logger = $logger;
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    protected function execute (InputInterface $input, OutputInterface $output) : ?int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title("Cron Jobs");
+
+        $now = new \DateTimeImmutable();
+        $jobFailed = false;
+
+        foreach ($this->registry->getAllJobs() as $job)
+        {
+            $io->section("Job: {$job->getName()}");
+
+            try
```

Comment:
Can't we combine the two `try-catch` blocks?

```php
            try
            {
                $wrappedJob = new WrappedJob($job, $now);

                if (!$this->logModel->isDue($wrappedJob))
                {
                    $io->writeln("<fg=yellow>Not due</>");
                    continue;
                }

				// …execution logic…
            }
            catch (\InvalidArgumentException $e)
            {
				// …
            }
            catch (\Exception $e)
            {
				// …
            }
```

Reply:
No, as the regular code can throw a `InvalidArgumentException` as well, but I want to handle it differently. Because if `execute` throws an `InvalidArgumentException` that's a bug in the code, not an invalid cron tab definition.

Reply:
Good point. So we just keep it like that.